src/v/kafka/server/group.h

```diff
@@ -587,6 +587,16 @@ class group {
         const group& _group;
     };
 
+    mutex& get_tx_lock(model::producer_id pid) { return _tx_locks[pid]; }
```

Comment:
I wonder if it's worth having something like:
```cpp
    template<typename Func>
    auto with_pid_lock(model::producer_id, Func&& func) {
        auto reply = co_await _tx_locks[pid].with(std::forward<Func>(func));
        gc_tx_lock(id);
        co_return reply;
    }
```

To be used, e.g.:
```cpp
        auto id = r.pid.get_id();
        co_return co_await with_pid_lock(
          id,
          [this, r = std::move(r)]() mutable {
              return abort_tx(std::move(r));
          }));
```

Reply:
Good idea, I've changed it!